src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientFetchFactory.java

```diff
@@ -247,9 +244,8 @@ public MessageConsumer newStream(
             }
 
             final KafkaFilterCondition condition = cursorFactory.asCondition(filters);
-            final KafkaAge maximumAge = filters.isEmpty() ||
-                filters.anyMatch(a -> !a.conditions().anyMatch(c -> c.kind() == KIND_AGE && c.age().get() == HISTORICAL)) ?
-                    LIVE : HISTORICAL;
+            final long latestOffset = kafkaFetchBeginEx.partition().latestOffset();
+            final KafkaOffsetType maximumAge = KafkaOffsetType.valueOf((byte) latestOffset);
```

Comment:
Please rename `maximumAge` to `maximumOffset` now that we have changed the type from `KafkaAge` to `KafkaOffsetType`.